ml_tto/automatic_emittance/utils.py

```diff
@@ -144,12 +150,204 @@ def calculate_bounding_box_penalty(
-            np.abs(
-                bounding_box_coordinates - center
-            ) - extent / 2
-        )
+        return np.max(np.abs(bounding_box_coordinates - center) - extent / 2)
 
     else:
         raise ValueError(f"ROI type {type(roi)} is not supported for ")
 
+
+def emittance_from_h5(
+    h5_filename: str, thin_lens=False, maxiter=None
+):
+    """
+    Parse HDF5 values needed for emittance calculation and calculate the emittance.
+
+    Parameters
+    ----------
+    h5_filename : str
+        The path to the HDF5 file.
+    thin_lens : bool, optional
+        Whether to use the thin lens approximation. Default is False.
+    maxiter : int, optional
+        Maximum number of iterations for the fitting optimization. Default is None.
+
+    Returns
+    -------
+    dict
+        The results of the emittance calculation.
+
+    Raises
+    ------
+    OSError
+        If the file is not found.
+    """
+    # Parses h5 values needed
+    if os.path.exists(h5_filename):
+        saver = H5Saver()
+        result_dict = saver.load_from_h5(filepath=h5_filename)
+    else:
+        raise OSError(f"File {h5_filename} is not found.")
+
+    inputs = {
+        "quad_vals": result_dict["quadrupole_pv_values"],  # dict, in kG
+        "beamsizes": result_dict["rms_beamsizes"],  # dict, in m
+        "q_len": result_dict["metadata"]["magnet"]["metadata"]["l_eff"],  # in m
+        "rmat": result_dict["metadata"]["rmat"],  # 2D array
+        "energy": result_dict["metadata"]["energy"],  # in eV
+    }
+
+    if (
+        "design_twiss" in result_dict["metadata"]
+        and result_dict["metadata"]["design_twiss"] is not None
+    ):
+        twiss_design = np.array(
+            [
+                [
+                    result_dict["metadata"]["design_twiss"]["beta_x"],
+                    result_dict["metadata"]["design_twiss"]["alpha_x"],
+                ],
+                [
+                    result_dict["metadata"]["design_twiss"]["beta_y"],
+                    result_dict["metadata"]["design_twiss"]["alpha_y"],
+                ],
+            ]
+        )
+    else:
+        twiss_design = None
+    inputs["twiss_design"] = twiss_design
+    inputs["thin_lens"] = thin_lens
+    inputs["maxiter"] = maxiter
+
+    # Call wrapper that takes quads in machine units and beamsize in meters
+    results = compute_emit_bmag_machine_units(**inputs)
+    results["metadata"] = result_dict["metadata"]
+
+    return EmittanceMeasurementResult(**results)
+
+
+def compute_emit_bmag_machine_units(
+    quad_vals: dict,
+    beamsizes: dict,
+    q_len: float,
+    rmat: np.ndarray,
+    energy: float,
+    twiss_design: np.ndarray,
+    thin_lens: bool = False,
+    maxiter: int = None,
+):
+    """
+    Wrapper for compute_emit_bmag that takes quads in machine units and beamsize in meters.
+
+    Parameters
+    ----------
+    quad_vals : dict
+        A dict containing the quadrupole values in kG for x and y respectively.
```

Comment:
can we use lists for this instead of dicts? Its not clear what the keys should be and we assume in the for loop that this contains 2 elements already

Reply:
same for the elements below